### ISSUE TITLE
Code Editor 2: enabling ligatures makes code wider

### ISSUE THREAD

**Issue by gnysek** [YoYoGames/GameMaker-Bugs#8397]
### Description
In CE2, when enabling ligatures, letters become closer to each other, even when there's no ligatures used

### Steps To Reproduce
1. Write some code
2. Go to Preferences > CE2 > Theme > *enable* Show Ligatures
3. Notice that each letter is 1px wider now, and some are so close, that they overlap (notice "m" below)

![Image](https://github.com/user-attachments/assets/6399c863-7fa5-404e-bfec-f35ccaeded24)
![Image](https://github.com/user-attachments/assets/d3d5864b-9081-4f2b-b249-e2b7ad180829)
both blended:
![Image](https://github.com/user-attachments/assets/60a35274-4c04-4439-8392-4d386bbc3e0f)

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.674 Runtime v2024.1100.0.700

### Which operating system(s) are you seeing the problem on?
Linux Ubuntu 22.04.5 LTS kernel 6.8.0.48

[edbec741-d34a-4c0c-970b-a1d5f0e615de](https://api.gamemaker.io/api/github/downloads/edbec741-d34a-4c0c-970b-a1d5f0e615de)

**Comment by gnysek** [comment]
Might be connected to #8097

**Comment by zreedy** [comment]
Appears fixed for `FiraCode` pt 10 font. Probably fixed as apart of changes made for https://github.com/YoYoGames/GameMaker-Bugs/issues/8097

**Comment by gnysek** [comment]
Yeah, could be connected to mentioned issue if that was happening on all non-ASCII characters (as ligatures/diacritics won't make much difference in that case).

**Comment by zreedy** [comment]
The renderer has a "simple" and a "complex" path where the simple path is taken if a string is entirely ASCII without ligatures. So enabling ligatures forces the "complex" path to be taken in most cases. The bug addressed in 8097 was a difference between kerning calculations in the simple vs complex paths. As such it would affect the above strings where `//` is a ligature (at least in FiraCode)

**Comment by zreedy** [comment]
Fixed with https://github.com/YoYoGames/GameMaker-Bugs/issues/8097

**Comment by Emc1923** [comment]
Closing as per comment, thanks.
